### PR TITLE
APIS-7255 Add new create deskpro ticket endpoint

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
-# api-platform-deskpro
+# API Platform Deskpro
 
-This is a placeholder README.md for a new repository
+Service for the creation of Deskpro Horizon tickets for use by API Platform
```
Comment (johnsgp) [inline README.md:4]: Perhaps `Service for interfacing with Deskpro Horizon for use by API Platform`. The next step is to interface to organizations, so will someone remember to edit the README at that point?

File: app/uk/gov/hmrc/apiplatformdeskpro/domain/models/CreateTicketRequest.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    name: String,
+    email: String
+  )
+
+case class CreateTicketRequest(
+    person: DeskproPerson,
+    subject: String,
+    message: String,
+    apiName: Option[String],
+    applicationId: Option[String],         // Leaving as a string as that's what's in DevHub
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apiplatformdeskpro/domain/models/CreateTicketRequest.scala:31]: Is the comment necessary? This service should match the types of the Deskpro API, so String seems fine without an explanation. Same with the team member email address.

File: app/uk/gov/hmrc/apiplatformdeskpro/domain/models/connector/DeskproTicket.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    name: String,
+    email: String
+  )
+
+case class DeskproTicketMessage(
+    message: String,
+    format: String = "html"
+  )
+
+// object DeskproTicketMessage {
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apiplatformdeskpro/domain/models/connector/DeskproTicket.scala:31]: Is this commented code needed?

**Review by johnsgp — DISMISSED** [review]

**Comment by johnsgp** [comment]
It's been helpful having `-Ddeskpro.api-key=${DESKPRO_KEY}` in run_local.sh of third-party-developer-frontend for local testing, so you can set an environment variable with your key.

File: README.md (hunk 1 above)
Comment (petekirby-ee) [reply README.md]: Done

File: app/uk/gov/hmrc/apiplatformdeskpro/domain/models/CreateTicketRequest.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformdeskpro/domain/models/CreateTicketRequest.scala]: Removed

File: app/uk/gov/hmrc/apiplatformdeskpro/domain/models/connector/DeskproTicket.scala (hunk 3 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformdeskpro/domain/models/connector/DeskproTicket.scala]: Yes, good point. Fixed.

**Review by johnsgp — APPROVED** [review]

**Pull request merged**